google/cloud/opentelemetry/internal/recordable_test.cc

```diff
@@ -242,6 +270,30 @@ TEST(Recordable, SetAttributeRespectsLimit) {
   EXPECT_EQ(attributes.dropped_attributes_count(), 1);
 }
 
+TEST(Recordable, SetStartTime) {
+  auto start = std::chrono::system_clock::now();
+  auto expected = absl::FromChrono(start);
+
+  auto rec = Recordable(Project(kProjectId));
+  rec.SetStartTime(start);
+  auto proto = std::move(rec).as_proto();
+  auto actual = internal::ToAbslTime(proto.start_time());
+  EXPECT_EQ(actual, expected);
+}
+
+TEST(Recordable, SetDuration) {
+  auto start = std::chrono::system_clock::now();
+  auto duration = std::chrono::nanoseconds(12345);
+  auto expected = absl::FromChrono(start) + absl::FromChrono(duration);
```

Comment:
Here, for example, and in `SetDuration()`, it is important to do the addition in the `absl::Time` domain.